aiogram/types/update.py

```diff
@@ -53,3 +53,62 @@ class Update(TelegramObject):
     """*Optional*. The bot's chat member status was updated in a chat. For private chats, this update is received only when the bot is blocked or unblocked by the user."""
     chat_member: Optional[ChatMemberUpdated] = None
     """*Optional*. A chat member's status was updated in a chat. The bot must be an administrator in the chat and must explicitly specify 'chat_member' in the list of *allowed_updates* to receive these updates."""
+
+    @property
```

Comment:
Shouldn't we add `@lru_cache` also, as in 2.x now?
https://github.com/aiogram/aiogram/blob/3aa40224a23f4cd0d8b824c07ebefceae913be07/aiogram/types/message.py#L108-L110

Reply:
I got `TypeError: unhashable type: 'Update'` because Pydantic BaseModel is unhashable
Reference https://github.com/samuelcolvin/pydantic/issues/1303

What do you think about possibility of adding hashing to TelegramObject? It seems to me that this is unlikely, since TelegramObject may have unhashable objects, like list[MessageEntity]

I also think that the type of update will not be checked as often as the content_type of message.
Now we have only one check of the update type in thein Dispatcher, I also think this check can be used in middlewares.

Reply:
last, first - content type position in check chain

```
Run #1
Call times: 1_000_000
Update type last     Cached 0:00:00.221997
Update type last     Ucached 0:00:00.380021
------------------------------------------
Update type first    Cached 0:00:00.221000
Update type first    Ucached 0:00:00.130010
------------------------------------------
Message type last    Cached 0:00:00.290000
Message type last    Ucached 0:00:00.909990
------------------------------------------
Message type first   Cached 0:00:00.296010
Message type first   Ucached 0:00:00.191000

Run #2
Call times: 1_000_000
Update type last     Cached 0:00:00.219978
Update type last     Ucached 0:00:00.382022
--------------------------------------------
Update type first    Cached 0:00:00.220000
Update type first    Ucached 0:00:00.131000
-------------------------------------------
Message type last    Cached 0:00:00.294010
Message type last    Ucached 0:00:00.898993
--------------------------------------------
Message type first   Cached 0:00:00.303007
Message type first   Ucached 0:00:00.190990
```

Reply:
> Shouldn't we add `@lru_cache` also, as in 2.x now?

@evgfilim1 I still can't imagine what you wanna cache on Update detecting. Every update is unique in the most cases. Please provide more info about your suggestion

Reply:
@darksidecat `functools.cache` is incompatible with Python 3.8, but aiogram 3.0 should support 3.8–3.10

@Olegt0rr After this PR, our users might use this newly introduced field for their own purposes, so caching can help improve speed a bit.

Reply:
@evgfilim1 
I could not get a sufficient performance improvement from using the cache, in the case when the field we need is at the beginning of the IF chain, we get a performance degradation.

Perhaps you can run the tests described above and see if it can be improved

```
Call times: 1_000_000
Update type last     Cached   0:00:00.221997
Update type last     Uncached 0:00:00.380021
------------------------------------------
Update type first    Cached   0:00:00.221000
Update type first    Uncached 0:00:00.130010
```